LocalPackages/AIChat/Sources/AIChat/Public API/AIChatViewController.swift

```diff
@@ -0,0 +1,204 @@
+        self.timerPixelHandler = TimerPixelHandler(pixelHandler: pixelHandler)
+
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+         fatalError("init(coder:) has not been implemented")
+     }
+}
+
+// MARK: - Lifecycle
+extension AIChatViewController {
+
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+        self.view.backgroundColor = .black
+
+        setupNavigationBar()
+
+        subscribeToCleanupPublisher()
+    }
+
+    public override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        addWebViewController()
+    }
+
+    public override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        timerPixelHandler.sendOpenPixel()
+        chatModel.cancelTimer()
+    }
+
+    public override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+
+        if viewIfLoaded?.window == nil {
+            chatModel.cancelTimer()
+            removeWebViewController()
+        }
+    }
+}
+
+// MARK: - Views Setup
+extension AIChatViewController {
+
+    private func setupNavigationBar() {
+        guard let navigationController = navigationController else { return }
+
+        let appearance = UINavigationBarAppearance()
+        appearance.configureWithTransparentBackground()
+        appearance.backgroundColor = .clear
+        appearance.shadowImage = UIImage()
+        appearance.shadowColor = .clear
+
+        navigationController.navigationBar.standardAppearance = appearance
+        navigationController.navigationBar.scrollEdgeAppearance = appearance
+        navigationController.navigationBar.compactAppearance = appearance
+        navigationController.navigationBar.isTranslucent = true
+
+        let imageView = UIImageView(image: UIImage(named: "Logo"))
+        imageView.contentMode = .scaleAspectFit
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+
+        let imageSize: CGFloat = 28
+        NSLayoutConstraint.activate([
+            imageView.widthAnchor.constraint(equalToConstant: imageSize),
+            imageView.heightAnchor.constraint(equalToConstant: imageSize)
+        ])
+
+        let titleLabel = UILabel()
+        titleLabel.text = UserText.aiChatTitle
+        titleLabel.font = UIFont.systemFont(ofSize: 17, weight: .semibold)
+        titleLabel.textColor = .white
+        let stackView = UIStackView(arrangedSubviews: [imageView, titleLabel])
+        stackView.axis = .horizontal
+        stackView.spacing = 8
+        stackView.alignment = .center
+        stackView.distribution = .fill
+
+        navigationItem.leftBarButtonItem = UIBarButtonItem(customView: stackView)
+
+        let closeButton = UIBarButtonItem(
+            image: UIImage(named: "Close-24"),
+            style: .plain,
+            target: self,
+            action: #selector(closeAIChat)
+        )
+        closeButton.tintColor = .white
```

Comment:
I think this is missing an accessibility identifier, so the asset name will get used as the label.